dojo/tools/kics/parser.py

```diff
@@ -62,6 +62,7 @@ def get_findings(self, filename, test):
                         + category
                         + issue_type
                         + file_name
+                        + expected_value
```

Comment:
the "similarity_id" seems to be unique and may suit better than the expected value, which is not.

I would suggest to add `similarity_id = item.get("similarity_id")` below the "acutal_value" and replace `expected_value` by `similarity_id` in the dupe_key

Reply:
I tested similarity_id and it will revert this fix again as in the above mentioned case the similarity_id is the same

![grafik](https://github.com/DefectDojo/django-DefectDojo/assets/47991713/a7fcdc0a-568f-4604-901a-25bc059e35c9)

Reply:
@quirinziessler do you have a unittestfile to help in this case? 
You can upload it here: https://github.com/DefectDojo/django-DefectDojo/issues/7966

Reply:
@manuel-sommer sorry for this, probably it was only unique for my output file. I noticed one finding being surpressed by your idea but it seems it is just a issue with the deduplication. Please keep the expected_value and adjust the hash_code in settings.dist.py. This together should solve the whole issue.
I would recommend to filter by "file_path, line, severity, description, title".